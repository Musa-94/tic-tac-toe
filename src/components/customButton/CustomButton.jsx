import React from 'react';
import './buttonStyles.less';

const CustomButton = props => {
    const {
        textTitle
    } = props

    return (
        <a href='/'
           children={textTitle}
           className={'border-button'}
        />
    )
}

export default React.memo(CustomButton);
