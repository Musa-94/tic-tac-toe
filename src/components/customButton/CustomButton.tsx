import React from 'react';
import { Button } from './styledComponents';

type TProps = {
    fontSize?: number,
    textColor?: string,
    textTitle?: string;
    borderColor?: string,
    borderWidth?: number,
    borderStyle?: string,
    backgroundColor?: string,
}

const CustomButton: React.FC<TProps> = props => {
    const {
        fontSize = 20,
        textTitle = 'Button',
        textColor,
        borderColor,
        borderWidth,
        borderStyle,
        backgroundColor,
    } = props;

    return (
        <Button type={'submit'}
                children={textTitle}
                fontSize={fontSize}
                textColor={textColor}
                borderColor={borderColor}
                borderWidth={borderWidth}
                borderStyle={borderStyle}
                backgroundColor={backgroundColor}
        />
    );
};

export default React.memo(CustomButton);
