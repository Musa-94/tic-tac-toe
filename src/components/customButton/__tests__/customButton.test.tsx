import React from 'react';
import CustomButton from '../CustomButton';

const props = {
    fontSize: 20,
    textTitle: 'Button',
    textColor: 'red',
    borderColor: 'black',
    backgroundColor: 'green'
};

describe('CustomButton snapshot', () => {
    it('should correctly render', () => {
        //@ts-ignore TODO remove when learn write to declare
        const component = renderer(<CustomButton {...props}/>);

        expect(component).toMatchSnapshot();
    });
});
