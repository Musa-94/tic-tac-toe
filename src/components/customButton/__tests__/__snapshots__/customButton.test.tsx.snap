// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton snapshot should correctly render 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  font-size: 20px;
  cursor: pointer;
  padding: 5px;
  color: red;
  border-color: black;
  background: green;
}

<button
  className="c0"
  fontSize={20}
  type="submit"
>
  Button
</button>
`;
